lib/ansible/modules/monitoring/datadog/datadog_monitor.py

```diff
@@ -235,6 +236,9 @@ def main():
     if not HAS_DATADOG:
         module.fail_json(msg=missing_required_lib('datadogpy'), exception=DATADOG_IMP_ERR)
 
+    if 'message' in module.params:
+        module.fail_json(msg="'message' is reserved keyword, please change this parameter to 'notification_message'")
```

Comment:
I don't think this would ever trigger. The `argument_spec` doesn't list `message`, so it would never be accepted to get to this point.

We would either have to allow it as an alias, or use `check_invalid_arguments=False`, which would potentially cause larger issues.

Reply:
This will be useful for users to migrate from 'message' to respective parameter argument. I can add alias 'message' in arg spec and set it on the path of deprecation.

Reply:
Yes, I agree with the intent, just without allowing `message` in the `argument_spec`, this path would never fire. You would get an invalid parameter error as is.

Reply:
Shouldn’t we had a control in AnsibleModule or any better place in Core that makes sure that « message » is not used in the future ?

As far as I know (could be wrong) there is no documented forbidden keys in argument_spec. So we should document this, maybe in the module dev guide ?

Reply:
@rrey I updated developer guide and add a check in `validate-modules` to check existence of restricted parameter name in module argument spec.